scopus-import/src/main/java/no/sikt/nva/scopus/conversion/PiaConnection.java

```diff
@@ -72,19 +74,20 @@ public PiaConnection() {
              new Environment());
     }
 
-    public URI getCristinPersonIdentifier(String scopusAuthorIdentifier) {
+    public Optional<URI> getCristinPersonIdentifier(String scopusAuthorIdentifier) {
         return attempt(() -> getPiaAuthorResponse(scopusAuthorIdentifier))
                    .map(this::getCristinNumber)
                    .map(Optional::orElseThrow)
                    .map(this::createCristinUriFromCristinNumber)
-                   .orElse(failure -> null);
+                   .toOptional();
     }
 
-    public URI getCristinOrganizationIdentifier(String scopusAffiliationIdentifier) {
+    public Optional<URI> getCristinOrganizationIdentifier(String scopusAffiliationIdentifier) {
         return attempt(() -> fetchAffiliationList(scopusAffiliationIdentifier))
                    .map(this::selectOneAffiliation)
                    .map(this::createCristinUriFromCristinOrganization)
-                   .orElse(failure -> null);
+                   .map(Optional::get)
```

Comment:
Looks weird…gets an optional optionally and returns an optional…is there a better way

Reply:
As long as there is never ever thrown an exception from this class that stops scopus-import I am happy.

Most authors or organizations with a scopus-identifiers, wont have a cristin-identifier.  
Add the stability of the PIA servers, and you will get why the "ignore all failures here" comment.

Now the reason this class exists is because the old scopus-import does this and it makes the importers job MUCH easier.